Shoko.Server/Providers/TraktTV/TraktTVHelper.cs

```diff
@@ -641,14 +641,27 @@ public void ScanForMatches()
                             }
                         }
 
-                        if (dictTraktSeasons != null && dictTraktSeasons.ContainsKey(xrefBase.TraktSeasonNumber))
+                        if (dictTraktSeasons != null && dictTraktSeasons.TryGetValue(xrefBase.TraktSeasonNumber, out var traktSeason))
                         {
-                            var episodeNumber = dictTraktSeasons[xrefBase.TraktSeasonNumber] +
-                                                (ep.EpisodeNumber + xrefBase.TraktStartEpisodeNumber - 2) -
-                                                (xrefBase.AniDBStartEpisodeNumber - 1);
-                            if (dictTraktEpisodes.ContainsKey(episodeNumber))
+                            int episodeNumber;
+                            {
+                                if (xrefBase.TraktStartEpisodeNumber == xrefBase.AniDBStartEpisodeNumber)
+                                {
+                                    // The Trakt and AniDB start episode numbers match
+                                    episodeNumber = (traktSeason - xrefBase.TraktStartEpisodeNumber ) + ep.EpisodeNumber;
+                                }
+                                
```

Comment:
Please remove the empty line between the if/else statements if you won't have a comment on the line. Functionality it works, but it's not in line with how we write if/else statements in this project.